instrumentation/src/com/intellij/rt/coverage/instrumentation/TouchCounter.java

```diff
@@ -38,9 +38,11 @@ public TouchCounter(MethodVisitor methodVisitor, BranchDataContainer branchData,
 
 
   public void visitLineNumber(int line, Label start) {
-    mv.visitVarInsn(Opcodes.ALOAD, getOrCreateLocalVariableIndex());
-    InstrumentationUtils.pushInt(mv, line);
-    mv.visitMethodInsn(Opcodes.INVOKESTATIC, ProjectData.PROJECT_DATA_OWNER, "trace", "(Ljava/lang/Object;I)V", false);
+    if (myBranchData.getContext().getLineData(line) != null) {
```

Comment:
how is this related?

Reply:
It fixes test traking for tracing mode: ignored lines (closing brackets) should not be covered.